rope/base/ast.py

```diff
@@ -5,7 +5,8 @@
 from rope.base import fscommands
 
 try:
-    from ast import _const_node_type_names
+    # Suppress the mypy complaint: Module "ast" has no attribute "_const_node_type_names"
+    from ast import _const_node_type_names  # type:ignore
```

Comment:
I'm not a fan of sprinkling so many `type: ignore` everywhere. Is there any way to tell mypy to just ignore untyped modules by default.

Reply:
@lieryan The `type: ignore` on line 8 of `rope.base.ast` is a special case. I don't know any way to remove it.

There are several ways of telling mypy to ignore untyped modules:
- This PR inserts a `type: ignore` *as the very first line* of `rope.base.prefs`.
- `.mypy` config files supposedly can suppress checks to specific modules.
  However, this capability seems buggy. It seems to suppress initial checks,
  but inferences in other files seem to "bleed into" the supposedly suppressed modules.

One alternative to `type: ignore` is annotating something as `Any`. However, this must substitutes one suppression pattern for another. Let's look at the `type: ignore` statements in this PR:
- Suppress warnings about optional imports. I am not bothered by those suppressions.
  These are the only suppression in `fscommands`, `doa`, `runmod`, `numpydocstrings`, and `project`.
- `oi.type_hinting_evaluate`: suppress (valid) mypy complaints about the `@multi` decorator.
  I haven't looked at the decorated functions in detail, I don't understand why `@multi` is needed,
  so I can't suggest any alternative. My guess: `type: ignore` is likely the only way.
- I know of no easy way to avoid the `type: ignore` in `contrib.autoimport.sqlite`,
  but I didn't try very hard&mdash;this file doesn't interest me at present.
- Afaik, the `type: ignore` suppressions in `refactor.move` can only be removed by "splitting" a name which 
  is assigned values with different types.
  This coding style becomes natural when using mypy. In my experience it always clarifies the code.
  But this PR avoids substantive changes to the code.

**Summary**

Most `type: ignore` suppressions seem harmless. The others seem unavoidable.

I'm not bothered by the `type: ignore` comments. There aren't that many, and I've carefully documented all the unusual suppressions.